src/scheduler/resv_info.cpp

```diff
@@ -1182,27 +1182,20 @@ check_new_reservations(status *policy, int pbs_sd, resource_resv **resvs, server
 								nresv_copy->select = new selspec(*nresv_copy->resv->select_standing);
 							}
 						}
-						/* Duplication deep-copies node info array. This array gets
-						 * overwritten and needs to be freed. This is an alternative
-						 * to creating another duplication function that only duplicates
-						 * the required fields.
-						 */
-						release_nodes(nresv_copy);
-
-						nresv_copy->resv->orig_nspec_arr = parse_execvnode(occr_execvnodes_arr[j], sinfo, nresv_copy->select);
-						nresv_copy->nspec_arr = combine_nspec_array(nresv_copy->resv->orig_nspec_arr);
-						nresv_copy->ninfo_arr = create_node_array_from_nspec(nresv_copy->nspec_arr);
-						nresv_copy->resv->resv_nodes = create_resv_nodes(nresv_copy->nspec_arr, sinfo);
 					}
+					release_nodes(nresv_copy);
```

Comment:
I just want to confirm, in the first iteration of the loop, we set nresv_copy = sinfo->resvs[i];, by taking release_nodes() outside the if-else block, we'll call release_nodes() on  sinfo->resvs[i] for the first iteration. That won't cause any issues right?

Reply:
The first time this reservation is confirmed, release_nodes() will do nothing.  We then copy in where the reservation will run.  If we reconfirm the reservation, its location is moving, so we need to clear what is there and copy in the new location.  So no, this won't cause any problems.